### PR TITLE
Rover: allow WATER_DEPTH mavlink message rate to be specified

### PR DESCRIPTION
**Pull request by peterbarker**
... and reduce the default rate

this is currently unconditionally streamed at 50Hz, chewing up all available bandwidth on some telemetry radios.

### CODE REVIEW

**Comment by rmackay9** [comment]
Here's the original PR that added support https://github.com/ArduPilot/ardupilot/pull/17804

**Comment by rmackay9** [comment]
Hi @peterbarker,

This looks good, what testing has been done?

**Comment by peterbarker** [comment]
> Hi @peterbarker,
> 
> This looks good, what testing has been done?

I found a convenient autotest, `Rover.DepthFinder` which seems to do all the testing required.  It actually picked up the fact my original patch wasn't filtering it to send only for boat frames!

I also did a general size-check for sanity:
```
Board               AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       *      *           *       *                 *      -32    *
Hitec-Airspeed      *                                                                     
KakuteH7-bdshot                *      *           *       *                 *      8      *
MatekF405                      *      *           *       *                 *      8      *
Pixhawk1-1M-bdshot             *                  *       *                 *      0      *
f103-QiotekPeriph   *                                                                     
f303-Universal      *                                                                     
iomcu                                                           *                         
revo-mini                      *      *           *       *                 *      8      *
skyviper-v2450                                    *                                       
```

**Review by rmackay9 — APPROVED** [review]

File: Rover/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -185,6 +185,65 @@ void GCS_MAVLINK_Rover::send_rangefinder() const
...
+    if (!rangefinder->has_orientation(ROTATION_PITCH_270)) {
+        return;
+    }
+
+    // get position
+    const AP_AHRS &ahrs = AP::ahrs();
+    Location loc;
+    IGNORE_RETURN(ahrs.get_location(loc));
+
+    for (uint8_t i=0; i<rangefinder->num_sensors(); i++) {
```
Comment (IamPete1) [inline Rover/GCS_Mavlink.cpp:216]: Should we cycle backends? Before this fix I have have the link be completely flooded on a vehicle with several rangefinders.

Comment (peterbarker) [reply Rover/GCS_Mavlink.cpp:216]: Yes, we should.

See last commit on here: https://github.com/ArduPilot/ardupilot/pull/28877

**Comment by peterbarker** [comment]
This PR fixed the partner's problem.

File: Rover/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -1206,3 +1273,6 @@ uint8_t GCS_MAVLINK_Rover::send_available_mode(uint8_t index) const
 
     return mode_count;
 }
+
+#if AP_RANGEFINDER_ENABLED
+#endif
```
Comment (IamPete1) [inline Rover/GCS_Mavlink.cpp:1278]: Left over from a rebase?

Comment (peterbarker) [reply Rover/GCS_Mavlink.cpp]: Not in this case - I moved the send-water-depth-code to be next to the send-rangefinder-code as the `#if`s already existed and the methods are related....

Removed these lines.

**Pull request merged**